index.js

```diff
@@ -85,36 +87,52 @@ exports.logUnsupportedVersion = function (napiVerison, log) {
  * Issues warnings to the supplied log for those N-API versions not supported
  * by the N-API runtime or the package.
  *
- * Note that this function is specific to the `prebuild` and `prebuild-install`
- * packages.
+ * Note that this function is specific to the
+ * [`prebuild`](https://github.com/prebuild/prebuild#prebuild) package.
+ *
+ * `target` is the list of targets to be built and is determined in one of
+ * three ways from the command line arguments:
+ * (1) `--target` specifies a specific target to build.
+ * (2) `--all` specifies all N-API versions supported by the package.
+ * (3) Neither of these specifies to build the single "best version available."
  *
- * @param {(Array<string>|string)} target The N-API version(s) to check.
- * @param {Object} prebuild An object created by the `prebuild` and
- * `prebuild-install` packages.
+ * `prebuild` is an array of objects in the form `{runtime: 'napi': target: '2'}`.
+ * The array contains the list of N-API versions that are supported by both the
+ * package being built and the currently running Node instance.
+ *
+ * The objective of this function is to issue a warning for those items that appear
+ * in the `target` argument but not in the `prebuild` argument.
+ * If a specific target is supported by the package (`packageSupportsVersion`) but
+ * but note in `prebuild`, the assumption is that the target is not supported by
+ * Node.
+ *
+ * @param {(Array<string>|string)} target The N-API version(s) to check. Target is
+ * @param {Object} prebuild A config object created by the `prebuild` package.
```

Comment:
I think you can document individual fields of an object with JSDoc:

```javascript
/**
 * @param {Object} prebuild A config object created by the `prebuild` package.
 * @param {string} prebuild.runtime The runtime
 * @param {string} prebuild.target The N-API version
 */

Reply:
My bad. It'a actually an array of objects.